mono/mini/trace.c

```diff
@@ -379,52 +347,47 @@ mono_trace_leave_method (MonoMethod *method, ...)
 				printf ("[INT32:%p:%d]", o, *((gint32 *)((char *)o + sizeof (MonoObject))));	
 			} else if  (o->vtable->klass == mono_defaults.int64_class) {
 				printf ("[INT64:%p:%lld]", o, (long long)*((gint64 *)((char *)o + sizeof (MonoObject))));	
-			} else
+			} else if (o->vtable->klass == mono_defaults.string_class) {
+				char *as;
+
+				as = string_to_utf8 ((MonoString*)o);
+				printf ("[STRING:%p:%s]", o, as);
+			} else {
 				printf ("[%s.%s:%p]", o->vtable->klass->name_space, o->vtable->klass->name, o);
-		} else
+			}
+		} else {
 			printf ("[OBJECT:%p]", o);
-	       
-		break;
-	}
-	case MONO_TYPE_PTR:
-	case MONO_TYPE_FNPTR:
-	case MONO_TYPE_ARRAY:
-	case MONO_TYPE_SZARRAY: {
-		gpointer p = va_arg (ap, gpointer);
-		printf ("result=%p", p);
+		}
 		break;
 	}
 	case MONO_TYPE_I8: {
-		gint64 l =  va_arg (ap, gint64);
+		gint64 l =  *arg_in_stack_slot (buf, gint64);
 		printf ("lresult=0x%16llx", (long long)l);
 		break;
 	}
 	case MONO_TYPE_U8: {
-		gint64 l =  va_arg (ap, gint64);
+		gint64 l =  *arg_in_stack_slot (buf, gint64);
 		printf ("lresult=0x%16llx", (long long)l);
 		break;
 	}
 	case MONO_TYPE_R4:
 	case MONO_TYPE_R8: {
-		double f = va_arg (ap, double);
+		double f = *arg_in_stack_slot (buf, double);
 		printf ("FP=%f", f);
 		break;
 	}
-	case MONO_TYPE_VALUETYPE:  {
-		guint8 *p = (guint8 *)va_arg (ap, gpointer);
-		int j, size, align;
-		size = mono_type_size (type, &align);
+	case MONO_TYPE_VALUETYPE: {
 		printf ("[");
-		for (j = 0; p && j < size; j++)
-			printf ("%02x,", p [j]);
+		// FIXME:
```

Comment:
Is the call context code providing broken values here, or is it just a to-do for printing value types in a better way than raw bytes?